server/src/internalClusterTest/java/org/opensearch/ratelimitting/admissioncontrol/SimpleEnableAdmissionControlIT.java

```diff
@@ -0,0 +1,150 @@
+
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 2, numClientNodes = 1)
+public class SimpleEnableAdmissionControlIT extends OpenSearchIntegTestCase {
+
+    public static final Settings settings = Settings.builder()
+        .put(AdmissionControlSettings.ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), "enforced")
+        .put(CPUBasedAdmissionControllerSettings.CPU_BASED_ADMISSION_CONTROLLER_TRANSPORT_LAYER_MODE.getKey(), "enforced")
+        .build();
+
+    public static final String INDEX_NAME = "test_index";
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder().put(super.nodeSettings(nodeOrdinal)).put(settings).build();
+    }
+
+    public void testAdmissionControlRejectionFlow() {
+        assertAcked(
+            prepareCreate(
+                INDEX_NAME,
+                Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+            )
+        );
+        ensureGreen(INDEX_NAME);
+
+        Tuple<String, String> primaryReplicaNodeNames = getPrimaryReplicaNodeNames(INDEX_NAME);
+        String primaryName = primaryReplicaNodeNames.v1();
+        String replicaName = primaryReplicaNodeNames.v2();
+        String coordinatingOnlyNode = getCoordinatingOnlyNode();
+
+        AdmissionControlService admissionControlServicePrimary = internalCluster().getInstance(AdmissionControlService.class, primaryName);
+        AdmissionControlService admissionControlServiceReplica = internalCluster().getInstance(AdmissionControlService.class, replicaName);
+        final BulkRequest bulkRequest = new BulkRequest();
+        for (int i = 0; i < 3; ++i) {
+            IndexRequest request = new IndexRequest(INDEX_NAME).id(UUIDs.base64UUID())
+                .source(Collections.singletonMap("key", randomAlphaOfLength(50)));
+            bulkRequest.add(request);
+        }
+
+        // evaluating the admission control triggered or not
+        ActionFuture<BulkResponse> successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        successFuture.actionGet();
+        assertEquals(admissionControlServicePrimary.getAdmissionControllers().size(), 1);
+        assertEquals(admissionControlServicePrimary.getAdmissionControllers().get(0).getRejectionCount(), 1);
+        assertEquals(admissionControlServiceReplica.getAdmissionControllers().size(), 1);
+        assertEquals(admissionControlServiceReplica.getAdmissionControllers().get(0).getRejectionCount(), 0);
+
+        // evaluating the rejection to increment
+        successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        successFuture.actionGet();
+        assertEquals(admissionControlServicePrimary.getAdmissionControllers().size(), 1);
```

Comment:
Can we add a `TODO` here. As I understand, with rejection logic integration the test will need updation.

Reply:
Updated.